gammapy/analysis/config.py

```diff
@@ -0,0 +1,189 @@
+from astropy.coordinates import Angle
+from astropy.time import Time
+from astropy.units import Quantity
+from gammapy.utils.scripts import make_path, read_yaml
+from pydantic import BaseModel, FilePath, validator
+from pydantic.utils import deep_update
+from pathlib import Path
+from typing import List
+from enum import Enum
+import yaml
+
+__all__ = ["Config", "General"]
+
+
+class AngleType(Angle):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return AngleType(v)
+
+
+class EnergyType(Quantity):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        assert isinstance(Quantity(v).to("erg"), Quantity)
```

Comment:
Suggest to write this:
```
v = Quantity(v)
assert v.unit.physical_type == "energy"
return v
```
The main advantage is that this converts to a Quantity once, whereas the current code does it twice.

Reply:
Done.